### PR TITLE
WPF ARM64: Update WPF Arcade SDK to support ARM64

### PR DESCRIPTION
**Pull request by ryalanms**
This PR includes changes to the WPF Arcade SDK, both SDK props/targets and source files that ship with the WPF Arcade SDK.  

The WPF Arcade SDK is consumed as a package dependency by WPF's internal repo.  The WPF Arcade SDK updates must go in to the public WPF repo before the native ARM64 changes can be checked-in to the internal repo.  

After this PR is submitted, verify the associated DARC update has been merged in WPF internal before merging the internal ARM64 changes.  

Co-authored-by: Alexandre Zollinger Chohfi <...>
Co-authored-by: Marcpems <...>
Co-authored-by: Juan Sebastian Hoyos Ayala <...>
Co-authored-by: Ryland <41491307+ryalanms@users.noreply.github.com>

#4034

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/WpfGfx/core/hw/ShaderAssemblies.targets [hunk 1]
```diff
@@ -19,8 +19,11 @@
     <ResourceFileOutputPath>$(ResourceFileIncludePath)\ShaderAssemblies</ResourceFileOutputPath>
   </PropertyGroup>
   
+  <!-- To support building ARM64 on x86/x64, this must use x64 for ARM64 -->
   <PropertyGroup>
-    <FXCTOOL>$(Windows10SdkPath)bin\$(WindowsTargetPlatformVersion)\$(Architecture)\fxc.exe</FXCTOOL>
+    <FXCTOOLArchitecture Condition="'$(Architecture)'=='ARM64'">x64</FXCTOOLArchitecture>
```
Comment (ryalanms) [inline src/Microsoft.DotNet.Wpf/src/WpfGfx/core/hw/ShaderAssemblies.targets:24]: This file is also not included in the WPF Arcade SDK and should be removed.

File: src/Microsoft.DotNet.Wpf/src/WpfGfx/core/common/utils.cpp [hunk 2]
```diff
@@ -129,20 +129,27 @@ UINT GetPaddedByteCount(UINT cbSize)
...
 #error Exactly one of _X86_, _AMD64_, _ARM_ should be defined
 #endif 
+*/
 
 #if defined(_X86_) 
     return cbSize;
 #elif defined(_AMD64_)
     return cbSize;
-#elif defined(_ARM_)
+// TODO: ARM64PORT: FIX THIS
```
Comment (azchohfi) [inline src/Microsoft.DotNet.Wpf/src/WpfGfx/core/common/utils.cpp]: Is comment this stale?

**Review by azchohfi — APPROVED** [review]

File: eng/WpfArcadeSdk/tools/Wpf.Cpp.targets [hunk 3]
```diff
@@ -401,11 +401,11 @@ using namespace System::Runtime::Versioning;
     </ItemGroup>
     <MSBuild Projects="@(ProjectToBuild)"
              Targets="Restore;ResolveReferences;IdentifyNugetReferences"
-             Properties="TargetFramework=$(TargetFramework);Platform=$(Architecture);RuntimeIdentifier=win-$(Architecture)">
+             Properties="TargetFramework=$(TargetFramework);Platform=$(Architecture);RuntimeIdentifier=win10-$(Architecture)">
```
Comment (hoyosjs) [inline eng/WpfArcadeSdk/tools/Wpf.Cpp.targets:404]: Why is this win10, as opposed to win?

File: eng/WpfArcadeSdk/tools/Packaging.targets [hunk 4]
```diff
@@ -301,13 +302,19 @@ $(PreparePackageAssetsDependsOn):
       <RootFolderContents Include="$(ArtifactsPackagingDir)$(NormalizedPackageName)\*.*"/>
     </ItemGroup>
 
+    <PropertyGroup>
+      <DestinationRuntimeSubFolder Condition="'$(Platform)'=='AnyCPU' or '$(Platform)'=='x86' or '$(Platform)'=='Win32'">runtimes\win-x86\native\</DestinationRuntimeSubFolder>
+      <DestinationRuntimeSubFolder Condition="'$(Platform)'=='x64'">runtimes\win-x64\native\</DestinationRuntimeSubFolder>
+      <DestinationRuntimeSubFolder Condition="'$(Platform)'=='arm64'">runtimes\win-arm64\native\</DestinationRuntimeSubFolder>
```
Comment (hoyosjs) [inline eng/WpfArcadeSdk/tools/Packaging.targets:308]: Why do we redefine this if `IdentifyPackageAssets` already does this? Or should this be hoisted into a common target?

Comment (ryalanms) [reply eng/WpfArcadeSdk/tools/Packaging.targets:308]: I believe these paths represent different things:  non-runtime specific binaries (`DestinationSubFolder`) and runtime-specific binaries (`DestinationRuntimeSubFolder`).  `DestinationSubFolder` should only be set to the same value as `DestinationRuntimeSubFolder` for native components, not for all components.

Comment (hoyosjs) [reply eng/WpfArcadeSdk/tools/Packaging.targets:308]: I misread. LGTM on this part.

Comment (ryalanms) [reply eng/WpfArcadeSdk/tools/Packaging.targets:308]: Thanks.  I'm pushing one more change with the TFM update (win10-arm64 -> win-arm64).

Comment (ryalanms) [reply eng/WpfArcadeSdk/tools/Packaging.targets:308]: GitHub is incredibly sensitive about the co-author formatting.  Those lines must be flush left and at the end of the commit message or they are ignored.

File: eng/WpfArcadeSdk/tools/Wpf.Cpp.targets [hunk 5]
```diff
@@ -554,11 +558,11 @@ using namespace System::Runtime::Versioning;
...
     <MSBuild Projects="@(ProjectToBuild)" UnloadProjectsOnCompletion="true"
              Targets="Restore"
-             Properties="TargetFramework=$(TfmForBuild);Platform=$(Architecture);RuntimeIdentifier=win-$(Architecture);IsRestoring=true" />
-    
+             Properties="TargetFramework=$(TfmForBuild);Platform=$(Architecture);RuntimeIdentifier=win10-$(Architecture);IsRestoring=true" />
+
     <MSBuild Projects="@(ProjectToBuild)"
          Targets="IdentifyNuGetReferences"
-         Properties="TargetFramework=$(TfmForBuild);Platform=$(Architecture);RuntimeIdentifier=win-$(Architecture)">
+         Properties="TargetFramework=$(TfmForBuild);Platform=$(Architecture);RuntimeIdentifier=win10-$(Architecture)">
```
Comment (hoyosjs) [inline eng/WpfArcadeSdk/tools/Wpf.Cpp.targets:565]: @ryalanms this file also has 2 references to win10 in the RID

**Pull request merged**